clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/SearchableProvider.java

```diff
@@ -50,6 +54,42 @@
    */
   Map<String, String> parseKey(String key);
 
+  default Optional<KeyParser> getKeyParser() {
+    return Optional.empty();
+  }
+  /**
+   *
+   */
+  default String buildSearchTerm(String type, String queryTerm) {
+    String prefix = getKeyParser().map(KeyParser::getCloudProvider).orElse("*");
```

Comment:
We have a legitimate `KeyParser` w/o a cloud provider I guess?

Reply:
IMO more inline comments in here too

Reply:
getKeyParser itself is optional, most cloud provider Keys don't currently implement that interface

Reply:
👍